### PR TITLE
Update to newton-usd-schemas 0.3.1 and lock mujoco to 3.8.x

### PR DESCRIPTION
**Pull request by andrewkaufman**
## Summary

- Update `newton-usd-schemas` dependency to `>=0.3.1` and lock `mujoco` to `>=3.8.1,<3.9`
- Author new Newton schema attributes side-by-side with MjcPhysics equivalents:
  - `NewtonSiteAPI` on site prims (presence-only, alongside `MjcSiteAPI`)
  - `NewtonMassAPI` with `newton:inertia` compact tensor on bodies with full inertia matrices (preserves `physics:diagonalInertia`/`principalAxes` as fallback for apps that don't support the tensor formulation)
  - `NewtonMassAPI` with `newton:massModel = "shell"` on shell inertia collision geoms
  - `newton:contactStiffness` and `newton:contactDamping` on physics materials (computed from MuJoCo `solref` using standard/direct mode conversion)
- Material deduplication hash now includes ke/kd to correctly separate materials with different solref values

## Test plan

- [x] All 176 existing tests pass (`uv run poe test`)
- [x] Lint clean (`uv run poe lint`)
- [ ] Manual review of converted USD assets for correctness

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: mujoco_usd_converter/_impl/geom.py [hunk 1]
```diff
@@ -12,6 +12,17 @@
 __all__ = ["convert_geom", "get_geom_name"]
 
 
+def solref_to_stiffness_damping(solref) -> tuple[float, float]:
```
Comment (andrewkaufman) [inline mujoco_usd_converter/_impl/geom.py:15]: @Milad-Rakhsha-NV tagged you to specifically double check this math. Don't worry about the rest of the PR

File: mujoco_usd_converter/_impl/geom.py [hunk 2]
```diff
@@ -349,14 +366,15 @@ def acquire_physics_material(geom: mujoco.MjsGeom, data: ConversionData) -> UsdS
     sliding_friction = geom.friction[0]
     torsional_friction = geom.friction[1]
     rolling_friction = geom.friction[2]
-    material_hash = Gf.Vec3f(sliding_friction, torsional_friction, rolling_friction)
+    ke, kd = solref_to_stiffness_damping(geom.solref)
```
Comment (andrewkaufman) [inline mujoco_usd_converter/_impl/geom.py:372]: To clarify, this unconditionally authors `newton:contactStiffness` & `newton:contactDamping` on the material, even when `mjc:solref` is left at the mujoco schema default values (whereas we do not author `mjc:solref` on the shape in that case). The reasoning is that non-mujoco solvers won't be mjc schema aware & need to know the explicit ke/kd values.

@jcarius-nv this is what you're hoping for, correct?

File: mujoco_usd_converter/_impl/geom.py [hunk 3]
```diff
@@ -366,27 +384,34 @@ def create_physics_material(physics_materials: Usd.Prim, geom: mujoco.MjsGeom, d
...
     torsional_friction = material.GetPrim().GetAttribute("mjc:torsionalfriction").Get()
     rolling_friction = material.GetPrim().GetAttribute("mjc:rollingfriction").Get()
-    return Gf.Vec3f(sliding_friction, torsional_friction, rolling_friction)
+    ke = material.GetPrim().GetAttribute("newton:contactStiffness").Get()
+    kd = material.GetPrim().GetAttribute("newton:contactDamping").Get()
+    return (sliding_friction, torsional_friction, rolling_friction, ke, kd)
+
+
+def _material_hash_close(a: tuple[float, ...], b: tuple[float, ...], tol: float = 1e-6) -> bool:
+    return all(abs(x - y) < tol for x, y in zip(a, b))
```
Comment (LouRohanNV) [inline mujoco_usd_converter/_impl/geom.py]: A review found that there is an edge case where this doesn't quite detect all invalid cases. I submitted a fix and an edge case test here: https://github.com/LouRohanNV/mujoco-usd-converter/commit/ad552add09bd38abaac749f43ded2e28bbd59743

This is pretty nit picky, but take a look at it.

Comment (andrewkaufman) [reply mujoco_usd_converter/_impl/geom.py]: Seems liks a good fix, push it here

File: pyproject.toml [hunk 4]
```diff
@@ -7,9 +7,9 @@ license = {text = "Apache-2.0"}
 readme = "README.md"
 requires-python = ">=3.10,<3.13"
 dependencies = [
-    "mujoco>=3.8.1",
+    "mujoco>=3.8.1,<3.9",
```
Comment (LouRohanNV) [inline pyproject.toml:10]: I suppose we're going to handle the [3.9 API breaking changes](https://mujoco.readthedocs.io/en/3.9.0/changelog.html), contact `margin` and `gap` parameters in another PR?

Comment (andrewkaufman) [reply pyproject.toml:10]: Yes, exactly, in a later release. Newton 1.3 is explicitly not taking mujoco 3.9 to avoid that change for now. I expect it will come with Newton 1.4 (in about month).

Comment (andrewkaufman) [reply pyproject.toml:10]: I'm hoping to jump straight past 3.9 to 3.10 minimum version tbh... I'd like to get the revised minimal schemas so we don't have to double-author anymore.

**Review by LouRohanNV — DISMISSED** [review]

File: mujoco_usd_converter/_impl/geom.py (hunk 2 above)
Comment (jcarius-nv) [reply mujoco_usd_converter/_impl/geom.py:372]: That's a good question. It basically comes down the the question whether we want

1. mjc **default** translating to the respective **default** (and likely different) values of contact stiffness and damping in other solvers
2. mjc **default** translating to the equivalent stiffness/damping values in other solvers as Mujoco would have done it.

We're going with 2 here, which Is fine by me.

However, if we're now doing a roundtrip mjc->usd->mujoco in newton, what will it pick up? If we don't authormjc:solref it will be inferred back from stiffness and damping?!

**Comment by andrewkaufman** [comment]
~~Good catch Jan — you're right that the roundtrip would lose contact response without it. We now author `mjc:solref` as a custom `double[]` attribute on every physics material (always authored, even for the default `[0.02, 1.0]`). Newton's `SchemaResolverMjc._mjc_legacy_material_solref()` will pick it up on the MuJoCo solver path.~~

Above was a nonsense reply from my coding agent, the commit it pushed is the wrong approach... Sorry about that.

Also included Lou's edge case fix for material hash comparison (inf==inf and length check).

File: mujoco_usd_converter/_impl/geom.py (hunk 2 above)
Comment (andrewkaufman) [reply mujoco_usd_converter/_impl/geom.py:372]: Good catch — the solref→ke/kd conversion is lossy (mass-dependent parameterization vs force-space), so SolverMuJoCo falling through to material-level ke/kd would be a semantic change.

Fixed by always authoring `mjc:solref` on the collision shape, even when it matches the schema default. Newton's SHAPE-level resolver now always finds the native solref before falling through to material-level ke/kd.

File: mujoco_usd_converter/_impl/geom.py (hunk 1 above)
Comment (Milad-Rakhsha) [reply mujoco_usd_converter/_impl/geom.py:15]: yep looks good to me. 
The implicit assumption is that d_width = 1, full impedance d(r) = 1.

File: mujoco_usd_converter/_impl/geom.py (hunk 2 above)
Comment (jcarius-nv) [reply mujoco_usd_converter/_impl/geom.py:372]: sounds good to me, maybe leave a comment to make that clear if we ever re-visit that part of the code :)

Comment (LouRohanNV) [reply mujoco_usd_converter/_impl/geom.py:372]: I think a comment is warranted where we force-author `mjc:solref`.

I will admit that I'm confused about some of this discussion (having never toiled over any of this in newton). I'm bristling at how default values (not authored) can have such strong implications downstream, but I don't have any better ideas.

Comment (andrewkaufman) [reply mujoco_usd_converter/_impl/geom.py:372]: Added

**Review by LouRohanNV — APPROVED** [review]
